pkg/ctl/create/cluster.go

```diff
@@ -150,10 +150,12 @@ func newNodeGroupChecker(i int, ng *api.NodeGroup) error {
 		ng.SecurityGroups.WithShared = api.NewBoolTrue()
 	}
 
-	if ng.AllowSSH {
-		if ng.SSHPublicKeyPath == "" {
-			ng.SSHPublicKeyPath = defaultSSHPublicKey
-		}
+	if ng.AllowSSH && ng.SSHPublicKeyPath == "" {
+		ng.SSHPublicKeyPath = defaultSSHPublicKey
+	}
+
+	if ng.SSHPublicKeyPath != defaultSSHPublicKey {
```

Comment:
So I'm actually not sure what's supposed to happen here in the current state of things.
We should change the field from `string` to `*string`, so when it wasn't specified it would be `nil`.